core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -345,7 +347,11 @@ public ClusterState execute(ClusterState currentState) throws Exception {
                                 .setRoutingNumShards(routingNumShards);
                             // Set up everything, now locally create the index to see that things are ok, and apply
                             final IndexMetaData tmpImd = tmpImdBuilder.settings(actualIndexSettings).build();
-                            if (request.waitForActiveShards().resolve(tmpImd) > tmpImd.getNumberOfReplicas() + 1) {
+                            ActiveShardCount waitForActiveShards = request.waitForActiveShards();
+                            if (waitForActiveShards == ActiveShardCount.DEFAULT) {
+                                waitForActiveShards = WAIT_FOR_ACTIVE_SHARDS_SETTING.get(tmpImd.getSettings());
```

Comment:
document that it has an effect on index creation. I saw it later. All good.

> On 29 Jul 2016, at 18:33, Ali Beyad <notifications@github.com> wrote:
> 
> In core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java:
> 
> > @@ -345,7 +347,11 @@ public ClusterState execute(ClusterState currentState) throws Exception {
> >                                  .setRoutingNumShards(routingNumShards);
> >                              // Set up everything, now locally create the index to see that things are ok, and apply
> >                              final IndexMetaData tmpImd = tmpImdBuilder.settings(actualIndexSettings).build();
> > -                            if (request.waitForActiveShards().resolve(tmpImd) > tmpImd.getNumberOfReplicas() + 1) {
> > +                            ActiveShardCount waitForActiveShards = request.waitForActiveShards();
> > +                            if (waitForActiveShards == ActiveShardCount.DEFAULT) {
> > +                                waitForActiveShards = WAIT_FOR_ACTIVE_SHARDS_SETTING.get(tmpImd.getSettings());
> 
> document the index.write.wait_for_active_shards setting? its included in the PR for doc updates: e.g. https://github.com/elastic/elasticsearch/pull/19581/files#diff-e973e987673cceb34ca841a82facc508R322
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>